src/pybind/mgr/dashboard/CMakeLists.txt

```diff
@@ -4,12 +4,20 @@ install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
   PATTERN "frontend/*" EXCLUDE
   PATTERN ".*" EXCLUDE)
 
+include(ExternalProject)
+ExternalProject_Add(grafonnet-lib
+  DOWNLOAD_DIR ${PROJECT_SOURCE_DIR}
+  GIT_REPOSITORY "https://github.com/grafana/grafonnet-lib.git"
+  GIT_TAG "v0.1.0"
+  GIT_SHALLOW TRUE)
+ 
 if(WITH_MGR_DASHBOARD_FRONTEND)
   # build from source
   add_subdirectory(frontend)
   if(WITH_TESTS)
     include(AddCephTest)
-    add_tox_test(mgr-dashboard TOX_ENVS py3 lint check openapi-check)
+    include_directories(grafonnet-lib ${PROJECT_SOURCE_DIR}})
```

Comment:
please refer to https://cmake.org/cmake/help/latest/command/include_directories.html

this line does not make sense at all.

i'd add `grafonnet-lib` as a dependency of `tests`, so it can be prepared before running `ctest`. like
```cmake
add_dependencies(tests
  grafonnet-lib)
```

as an alternative, we can create a fixture for `run-tox-mgr-dashboard`, like
```cmake
if(WITH_TESTS)
  include(AddCephTest)
  add_tox_test(mgr-dashboard
    TOX_ENVS py3 lint check openapi-check grafonnet-check)
  set(ver 0.1.0)
  set(name grafonnet-lib)
  set(tar ${CMAKE_CURRENT_BINARY_DIR}/v${ver}.tar.gz)
  file(DOWNLOAD 
    https://github.com/grafana/grafonnet-lib/archive/refs/tags/v{ver}.tar.gz
    ${tar})
  add_test(
    NAME extract-${name}
    COMMAND ${CMAKE_COMMAND} -E tar xfz ${tar}
    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
  set_tests_properties(extract-${${name} PROPERTIES
    FIXTURES_SETUP ${name})
  add_test(
    NAME remove-${name}
    COMMAND ${CMAKE_COMMAND} -E remove_directory ${name}-${ver})
  set_tests_properties(remove-${name} PROPERTIES
    FIXTURES_CLEANUP ${name})
  set_property(
    TEST run-tox-mgr-dashboard
    APPEND_STRING
    PROPERTY FIXTURES_REQUIRED ${name})
```

Reply:
@tchaikov , It still gives the error while importing grafonnet-lib - 'couldn't open import "../../../../grafonnet-lib/grafonnet/grafana.libsonnet. Tried both the methods you suggested.

Reply:
#42623